docs/Analyzer Configuration.md

```diff
@@ -0,0 +1,46 @@
+# Analyzer Configuration
+
+Starting with version `2.6.3`, all the analyzer NuGet packages produced in this repo, including the FxCop Analyzers NuGet package, support _.editorconfig based analyzer configuration_. End users can configure the behavior of specific CA rule(s) OR all configurable CA rules by specifying supported key-value pair options in an `.editorconfig` file. You can read more about `.editorconfig` format [here](https://editorconfig.org/).
+
+## .editorconfig format
+Analyzer configuration options from an .editorconfig file are parsed into _general_ and _specific_ configuration options. General configuration enables configuring the behavior of all CA rules for which the provided option is valid. Specific configuration enables configuring each CA rule ID or CA rules belonging to each rule category, such as 'Naming', 'Design', 'Performance', etc. Our options are _case-insensitive_. Below are the supported formats:
+   1. General configuration option:
+      1. `dotnet_code_quality.OptionName = OptionValue`
+   2. Specific configuration option:
+      1. `dotnet_code_quality.RuleId.OptionName = OptionValue`
+      2. `dotnet_code_quality.RuleCategory.OptionName = OptionValue`
+
+For example, end users can configure the analyzed API surface for analyzers using the below `api_surface` option specification:
+   1. General configuration option:
+      1. `dotnet_code_quality.api_surface = public`
+   2. Specific configuration option:
+      1. `dotnet_code_quality.CA1040.api_surface = public`
+      2. `dotnet_code_quality.Naming.api_surface = public`
+
+## Enabling .editorconfig based configuration for a project
+1. Per-project .editorconfig file: End users can enable .editorconfig based configuration for individual projects by just copying the .editorconfig file with the options to the project root directory. In future, we plan to support hierarchical directory based configuration with an .editorconfig file at the solution directory, repo root directory or even individual document directories.
+2. Shared .editorconfig file: If you would like to share a common .editorconfig file between projects, say `<%PathToSharedEditorConfig%>\.editorconfig`, then you should add the following MSBuild property group and item group to a shared props file that is imported _before_ the FxCop analyzer props files (that come from the FxCop analyzer NuGet package reference):
+```
+  <PropertyGroup>
```

Comment:
what is the plan for migration once real thing is in for roslyn? will this approach still work once real thing is supported? or it will stop working once real thing is checked in.

Reply:
At that point, we will no longer need .editorconfig as additional file, so customers will not require any special steps mentioned in this section - they can just delete this from their props. We will also delete the special hook in our analyzer props that marks default .editorconfig in project root directory as additional file, as well as delete the editorconfig parser and just query the `AnalyzerOptions` on the analysis context to get the parsed key-value pairs.